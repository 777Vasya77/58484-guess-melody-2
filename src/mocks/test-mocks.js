export const questions = [
  {
    'type': `artist`,
    'song': {
      'artist': `Endless Love`,
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Dirty_Mac.mp3`
    },
    'answers': [
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
        'artist': `Quincas Moreira`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Endless_Love.jpg`,
        'artist': `Endless Love`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Devon_Church.jpg`,
        'artist': `Devon Church`
      }
    ]
  },
  {
    'type': `artist`,
    'song': {
      'artist': `Dan Lebowitz`,
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Parkside.mp3`
    },
    'answers': [
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Spazz_Cardigan.jpg`,
        'artist': `Spazz Cardigan`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Unicorn_Heads.jpg`,
        'artist': `Unicorn Heads`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Dan_Lebowitz.jpg`,
        'artist': `Dan Lebowitz`
      }
    ]
  },
  {
    'type': `genre`,
    'genre': `alternative`,
    'answers': [
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Bark.mp3`,
        'genre': `reggae`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Addis_Ababa.mp3`,
        'genre': `reggae`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Memories_Station_of_Tomorrow.mp3`,
        'genre': `alternative`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Blue_Whale.mp3`,
        'genre': `alternative`
      }
    ]
  },
  {
    'type': `artist`,
    'song': {
      'artist': `Jesse Gallagher`,
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Memories_Station_of_Tomorrow.mp3`
    },
    'answers': [
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/The_Mini_Vandals.jpg`,
        'artist': `The Mini Vandals`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Spazz_Cardigan.jpg`,
        'artist': `Spazz Cardigan`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Jesse_Gallagher.jpg`,
        'artist': `Jesse Gallagher`
      }
    ]
  },
  {
    'type': `artist`,
    'song': {
      'artist': `Density & Time`,
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Clean_Break.mp3`
    },
    'answers': [
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Unicorn_Heads.jpg`,
        'artist': `Unicorn Heads`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`,
        'artist': `Density & Time`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Spazz_Cardigan.jpg`,
        'artist': `Spazz Cardigan`
      }
    ]
  },
  {
    'type': `artist`,
    'song': {
      'artist': `John Deley and the 41 Players`,
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Bark.mp3`
    },
    'answers': [
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Unicorn_Heads.jpg`,
        'artist': `Unicorn Heads`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/John_Deley_and_the_41_Players.jpg`,
        'artist': `John Deley and the 41 Players`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`,
        'artist': `Density & Time`
      }
    ]
  },
  {
    'type': `genre`,
    'genre': `country`,
    'answers': [
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Weak_Knight.mp3`,
        'genre': `electronic`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Clean_Break.mp3`,
        'genre': `electronic`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Parkside.mp3`,
        'genre': `country`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Azure.mp3`,
        'genre': `electronic`
      }
    ]
  },
  {
    'type': `artist`,
    'song': {
      'artist': `Spazz Cardigan`,
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Nothin_Yet.mp3`
    },
    'answers': [
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/sextile.jpg`,
        'artist': `Sextiler`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Spazz_Cardigan.jpg`,
        'artist': `Spazz Cardigan`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Unicorn_Heads.jpg`,
        'artist': `Unicorn Heads`
      }
    ]
  },
  {
    'type': `artist`,
    'song': {
      'artist': `Unicorn Heads`,
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Acoustic_Circles.mp3`
    },
    'answers': [
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Unicorn_Heads.jpg`,
        'artist': `Unicorn Heads`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Jesse_Gallagher.jpg`,
        'artist': `Jesse Gallagher`
      },
      {
        'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Silent_Partner.jpg`,
        'artist': `Silent Partner`
      }
    ]
  },
  {
    'type': `genre`,
    'genre': `country`,
    'answers': [
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Nothin_Yet.mp3`,
        'genre': `alternative`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Parkside.mp3`,
        'genre': `country`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Dub_Spirit.mp3`,
        'genre': `reggae`
      },
      {
        'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Bark.mp3`,
        'genre': `reggae`
      }
    ]
  }
];

export const questionTypeArtist = {
  'type': `artist`,
  'song': {
    'artist': `Endless Love`,
    'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Dirty_Mac.mp3`
  },
  'answers': [
    {
      'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
      'artist': `Quincas Moreira`
    },
    {
      'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Endless_Love.jpg`,
      'artist': `Endless Love`
    },
    {
      'picture': `https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Devon_Church.jpg`,
      'artist': `Devon Church`
    }
  ]
};

export const questionTypeGenre = {
  'type': `genre`,
  'genre': `alternative`,
  'answers': [
    {
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Bark.mp3`,
      'genre': `reggae`
    },
    {
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Addis_Ababa.mp3`,
      'genre': `reggae`
    },
    {
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Memories_Station_of_Tomorrow.mp3`,
      'genre': `alternative`
    },
    {
      'src': `https://htmlacademy-react-2.appspot.com/guess-melody/static/music/Blue_Whale.mp3`,
      'genre': `alternative`
    }
  ]
};
