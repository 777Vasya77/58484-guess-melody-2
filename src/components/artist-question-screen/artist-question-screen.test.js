import React from 'react';
import renderer from 'react-test-renderer';
import {questionTypeArtist as question} from '~/mocks/test-mocks';
import ArtistQuestionScreen from '~/components/artist-question-screen/artist-question-screen';

it(`ArtistQuestionScreen component render correctly`, () => {
  const tree = renderer.create(
      <ArtistQuestionScreen
        question={question}
        screenIndex={0}
        onAnswer={() => {}}
        mistakes={3}
        gameTime={3}
      />,
      {createNodeMock: () => ({})}
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
