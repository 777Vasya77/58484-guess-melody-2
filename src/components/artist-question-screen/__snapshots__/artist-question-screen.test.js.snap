// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen component render correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
    </div>
  </div>
  <form
    className="game__artist"
    onChange={[Function]}
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        type="radio"
        value="{\\"picture\\":\\"https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg\\",\\"artist\\":\\"Quincas Moreira\\"}"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Quincas Moreira"
          className="artist__picture"
          src="https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
        />
        Quincas Moreira
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        type="radio"
        value="{\\"picture\\":\\"https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Endless_Love.jpg\\",\\"artist\\":\\"Endless Love\\"}"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Endless Love"
          className="artist__picture"
          src="https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Endless_Love.jpg"
        />
        Endless Love
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        type="radio"
        value="{\\"picture\\":\\"https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Devon_Church.jpg\\",\\"artist\\":\\"Devon Church\\"}"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Devon Church"
          className="artist__picture"
          src="https://htmlacademy-react-2.appspot.com/guess-melody/static/artist/Devon_Church.jpg"
        />
        Devon Church
      </label>
    </div>
  </form>
</section>
`;
